recipes/iceoryx/all/conanfile.py

```diff
@@ -0,0 +1,244 @@
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 14)
+        if os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration(
+                'Using Iceoryx on Windows currently just possible with "shared=False"')
+        if (compiler == "gcc" or compiler == "clang") and compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration(
+                'Using Iceoryx with gcc or clang on Linux requires "compiler.libcxx=libstdc++11"')
+        if os == "Linux" and compiler == "gcc" and version <= "5":
+            raise ConanInvalidConfiguration(
+                "Using Iceoryx with gcc on Linux requires gcc 6 or higher.")
+        if os == "Linux" and compiler == "gcc" and version == "6":
+            self.output.warn(
+                "Iceoryx package is compiled with gcc 6, it is recommended to use 7 or higher")
+            self.output.warn(
+                "GCC 6 will built with warnings.")
+        if compiler == "Visual Studio" and version < "16":
+            raise ConanInvalidConfiguration(
+                "Iceoryx is just supported for Visual Studio compiler 16 and higher.")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["TOML_CONFIG"] = self.options.toml_config
+        self._cmake.configure()
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(self._pkg_share)
+        tools.rmdir(self._pkg_cmake)
+        tools.mkdir(self._pkg_res)
+        if self.options.toml_config:
+            tools.rename(
+                os.path.join(self._pkg_etc, "roudi_config_example.toml"),
+                os.path.join(self._pkg_res, "roudi_config.toml")
+            )
+        tools.rmdir(self._pkg_etc)
+        for alias, aliased in self._target_aliases.items():
+            cmake_file = "conan-official-{}-targets.cmake".format(
+                aliased.replace("::", "_")
+            )
+            self._create_cmake_module_alias_targets(
+                os.path.join(
+                    self.package_folder,
+                    self._module_subfolder,
+                    cmake_file
+                ),
+                alias,
+                aliased
+            )
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "iceoryx"
+        self.cpp_info.names["cmake_find_multi_package"] = "iceoryx"
+        # platform component
+        self.cpp_info.components["platform"].name = "platform"
```

Comment:
The components are a little on the generic side

Perhaps something like?

```suggestion
        self.cpp_info.components["iceoryx_platform"].name = "platform"
```

Reply:
yes, it would be nice to fix it, and not use global `name` for all generators.